File: leetcode/1302.层数最深叶子节点的和.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func deepestLeavesSum(root *TreeNode) int {
    //广度优先搜索
    ans := 0
    q := []*TreeNode{root}
    for len(q) > 0 {
        m := len(q)
        tmp := 0
        for i := 0; i < m; i++ {
            t := q[0]
            q = q[1:]
            tmp += t.Val
            if t.Left != nil {
                q = append(q, t.Left)
            }
            if t.Right != nil {
                q = append(q, t.Right)
            }
        }
        ans = tmp
    }
    return ans
}

File: leetcode/1302.层数最深叶子节点的和.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def deepestLeavesSum(self, root: Optional[TreeNode]) -> int:
        # 层序优先遍历
        que=deque()
        que.append(root)
        res=0
        while que:
            n=len(que)
            res=0
            for i in range(n):
                ans=que.pop()
                if ans.left:
                    que.appendleft(ans.left)
                if ans.right:
                    que.appendleft(ans.right)
                res+=ans.val
        return res
                